### DISCUSSION TITLE
On the use of GPUs for time dependent Non Linear problem

### DISCUSSION THREAD

**Discussion by CINTROINI** [mfem/mfem#4488] [General]
Dear MFEM community, 

We are developing a new code based on MFEM to simulate the time evolution of multiphase multicomponent systems.
Thanks to MFEM features, our code benefits from efficient MPI parallelization. 

We are now planning to test the parallelization on GPUs (or GPU/CPU hybrids). 

Have you already tested this kind of implementation on time-dependent non-linear problems like Example 10, or at least on a non-linear problem solved by a Newton-like algorithm?  

Is GPU available for non-linear form (partial assembly...) ?

thanks for your feedback and advices. 

Clément

**Comment by CINTROINI** [comment]
Up

**Comment by CINTROINI** [comment]
Feedback from the community would be greatly appreciated. Thanks in advance. 
Clément

**Comment by vladotomov** [comment]
PA+GPUs for `NonlinearForms` has been done before.
See the class `PANonlinearFormExtension`.
It is used in `/miniapps/meshing/pmesh-optimizer.cpp`, here:
```
ParNonlinearForm a(pfespace);
if (pa) { a.SetAssemblyLevel(AssemblyLevel::PARTIAL); }
```

**Reply by CINTROINI** [reply]
Thanks for your answer. We'll start a preliminary test with this source code.
